### PR TITLE
Fix str/bytes issue in SMTP error handling

### PR DESCRIPTION
**Pull request by dannyroberts**
https://sentry.io/organizations/dimagi/issues/1812947590/?project=136860&query=is%3Aunresolved

##### SUMMARY
We don't normally notice this since it's in error handling that doesn't happen much, but right now the error handling itself fails with an unrelated str/bytes issue

### CODE REVIEW

File: corehq/ex-submodules/dimagi/utils/django/email.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def send_HTML_email(subject, recipient, html_content, text_content=None,
     except SMTPDataError as e:
         # If the SES configuration has not been properly set up, resend the message
         if (
-            "Configuration Set does not exist" in e.smtp_error
+            b"Configuration Set does not exist" in e.smtp_error
```
Comment (millerdev) [inline corehq/ex-submodules/dimagi/utils/django/email.py:110]: I'm guessing this was a real thing that was tested at some point. Maybe we should try to coerce `smtp_error` to `str` instead? Could work to do
```suggestion
            "Configuration Set does not exist" in repr(e.smtp_error)
```

Comment (dannyroberts) [reply corehq/ex-submodules/dimagi/utils/django/email.py:110]: `reprr is interesting; here I guess it's sort of short hand for "convert ascii to str and we kind of don't care how non-ascii bytes are converted". I like it; it's like 3% clever but 100% intuitive how it's going to work.

**Comment by proteusvacuum** [comment]
This also needs to be changed here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/tasks.py#L87

**Review by biyeun — APPROVED** [review]

File: corehq/ex-submodules/dimagi/utils/django/email.py (hunk 1 above)
Comment (millerdev) [reply corehq/ex-submodules/dimagi/utils/django/email.py]: The thing that made me choose `repr` was the fact that previously it had apparently worked. I'm assuming this error handling was a result of a real error that was seen at some time, and maybe it actually worked for that situation? This makes me think that sometimes `e.smtp_error` is a `str` and sometimes its `bytes`. The _right thing_ to do would be to get to the bottom of why (or if) the type is different sometimes. This is really a band-aid fix.

**Review by millerdev — APPROVED** [review]

**Pull request merged**